### ISSUE TITLE
Missing type in Checkout.SessionCreateParams

### ISSUE THREAD

**Issue by mike-hand** [stripe/stripe-node#2096]
### Describe the bug

types/Checkout is missing the property payment_method_data.

https://docs.stripe.com/api/checkout/sessions/create

### To Reproduce

Add a params const and set the type to Stripe.Checkout.SessionCreateParams

### Expected behavior

Type should be available to use.  Does work when using it, typescript flags it as not available and causes unit tests to fail (just an fyi).

### Code snippets

```JavaScript
const stripeParams: Stripe.Checkout.SessionCreateParams = {
  payment_method_data: { allow_redisplay: 'always' }
}
```

### OS

Web

### Node version

v20.13.0

### Library version

15.8.0 (latest)

### API version

2023-10-16

**Comment by remi-stripe** [comment]
@mike-hand Thanks for the report! We'll work on a fix

**Comment by ramya-stripe** [comment]
@mike-hand The `Stripe.Checkout.SessionCreateParams` type was recently updated to include the property `payment_method_data` you are looking for.
- [changelog entry](https://github.com/stripe/stripe-node/blob/master/CHANGELOG.md#1530---2024-04-18)
- The [PR with the change](https://github.com/stripe/stripe-node/pull/2069/files#diff-605ab2872d7b9dc1600dd02cd931029f67551f8303ba8c19d1185d9e208c53b9).

> Does work when using it, typescript flags it as not available and causes unit tests to fail 

Can you elaborate what you mean by "does work when using it"?

**Comment by mike-hand** [comment]
"Does work when using it"... even tough typescript complains about it, when I include the property through a work-around, it works when the form is built.  It's just that typescript doesn't see the type is the only issue and causes the unit test to fail if it's included within.
const stripeParams = {
    allow_promotion_codes: true,
    automatic_tax: { enabled: true },
    ui_mode: 'embedded',
    
   // payment_method_data: { allow_redisplay: 'always' }, //<-- typescript/unit test complains.
}

//work-around adds the property so we can use it.  This property still works.
stripeParams['payment_method_data] = {
  allow_redisplay: 'always'
}

**Comment by ramya-stripe** [comment]
Understood

I am not able to reproduce the issue though. When using the latest stripe package, I am getting all the relevant type hints in my editor
![Screenshot 2024-05-30 at 9 29 23 AM](https://github.com/stripe/stripe-node/assets/100975018/652e893d-aa15-41d2-af7c-ca074058868f)

![Screenshot 2024-05-30 at 9 29 49 AM](https://github.com/stripe/stripe-node/assets/100975018/a057f832-cf97-4876-a37b-ff7cba51608a)

My hunch is that you are getting an older version of the stripe package where `payment_method_data` was not yet added to `SessionCreateParams`.

Can you recheck the version of the stripe package that is being used in your project? Try looking at the `node_modules/stripe/package.json` file

**Comment by mike-hand** [comment]
It's marked as updated last month.  I checked before I submitted the ticket, maybe is was recently pushed?

**Comment by ramya-stripe** [comment]
The changes were released on April 18th in version 15.3.0 of the stripe package

**Comment by ramya-stripe** [comment]
@mike-hand Hoping that updating to the latest version of the package has resolved the problem for you. Closing this issue out
